core/core-impl/src/main/java/ai/timefold/solver/core/impl/heuristic/selector/move/generic/list/ListInitializeMove.java

```diff
@@ -0,0 +1,77 @@
+package ai.timefold.solver.core.impl.heuristic.selector.move.generic.list;
+
+import java.util.Objects;
+
+import ai.timefold.solver.core.api.score.director.ScoreDirector;
+import ai.timefold.solver.core.impl.domain.variable.descriptor.ListVariableDescriptor;
+import ai.timefold.solver.core.impl.heuristic.move.AbstractEasyMove;
+import ai.timefold.solver.core.impl.score.director.VariableDescriptorAwareScoreDirector;
+
+public final class ListInitializeMove<Solution_> extends AbstractEasyMove<Solution_> {
+
+    private final ListVariableDescriptor<Solution_> variableDescriptor;
+    private final Object planningValue;
+
+    public ListInitializeMove(ListVariableDescriptor<Solution_> variableDescriptor, Object planningValue) {
+        this.variableDescriptor = variableDescriptor;
+        this.planningValue = planningValue;
+    }
+
+    public Object getInitializedValue() {
+        return planningValue;
+    }
+
+    // ************************************************************************
+    // Worker methods
+    // ************************************************************************
+
+    @Override
+    public boolean isMoveDoable(ScoreDirector<Solution_> scoreDirector) {
+        return true;
+    }
+
+    @Override
+    protected void doMoveOnGenuineVariables(ScoreDirector<Solution_> scoreDirector) {
+        var variableDescriptorAwareScoreDirector = (VariableDescriptorAwareScoreDirector<Solution_>) scoreDirector;
+        variableDescriptorAwareScoreDirector.beforeListVariableElementInitialized(variableDescriptor, planningValue);
```

Comment:
Yes. Something needs to tell the score director that the element was initialized.
But you may be right that this is not the best use of a move. I'll rethink it.

(In fact, removing this would likely also get rid of one use of the `CompositeMove`.